### PR TITLE
Cleanup area shadow code and Macro + remove more Raytracing macro

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR
This PR do various cleanup for the shadow
- rename GetAreaLightAttenuation to GetRectAreaShadowAttenuation
- reanme SKIP_RASTERIZED_SHADOWS => SKIP_RASTERIZED_AREA_SHADOWS
- removed RASTERIZED_AREA_LIGHT_SHADOWS macro
- factor area sahdow code of Eye, Lit and Stacklit in single location LightEvaluation:
EvaluateShadow_RectArea(). This also fix various issue like shadow evaluation call when not needed
and some debug mode not working with area shadow
- Added a colored shadow option in shader config which define shadow to be float3 or float (SHADOW_TYPE). Apply this new type to all shadow (to support colored shadow in both raster and raytrace in the future)
- rename DirectionalShadowType to SHADOW_TYPE

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by anisunity** [comment]
looks fine to me

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl [hunk 1]
```diff
@@ -472,6 +471,50 @@ float EvaluateShadow_Punctual(LightLoopContext lightLoopContext, PositionInputs
...
+#if defined(SCREEN_SPACE_SHADOWS) && !defined(_SURFACE_TYPE_TRANSPARENT) && (SHADERPASS != SHADERPASS_VOLUMETRIC_LIGHTING)
+    if ((light.screenSpaceShadowIndex & SCREEN_SPACE_SHADOW_INDEX_MASK) != INVALID_SCREEN_SPACE_SHADOW)
+    {
+        shadow = GetScreenSpaceShadow(posInput, light.screenSpaceShadowIndex);
+    }
+    else
+#endif
+    if ((light.shadowIndex >= 0) && (light.shadowDimmer > 0))
+    {
+        // light.positionRWS now contains the Light vector.
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl:498]: This comment seems out of place? light.positionRWS is not referenced.

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl:498]: indeed, come from the original code, I will removed

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightEvaluation.hlsl]: fixed

File: com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ The shader function `SampleShadow_PCSS` now requires you to pass in an additiona
 
 From Unity 2020.2, due to the change of shadow map, the enum HDShadowFilteringQuality have been moved to HDShadowManager.cs and the variables ShaderConfig.s_DeferredShadowFiltering as well as the option ShaderOptions.DeferredShadowFiltering have been removed from the code as they have no impact anymore.
 
+From Unity 2020.2, a new option is available name ColoredShadow. It allow to control if the shadow will be chromatic or monochrome. ColoredShadow have a performance cost. ColoredShadow are the default and currently only work with Raytrace shadow.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md:44]: From Unity 2020.2, a new option is available named ColoredShadow. It allows you to control whether a shadow is chromatic or monochrome. ColoredShadow is enabled by default and currently only works with [Ray-traced shadows](Ray-Traced-Shadows.md). Note that colored shadows are more resource-intensive to process than standard shadows.

File: com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md [hunk 3]
```diff
@@ -72,6 +73,9 @@ BSDFData bsdfData = ConvertSurfaceDataToBSDFData(posInput.positionSS, surfaceDat
 PreLightData preLightData = GetPreLightData(V, posInput, bsdfData);
 ```
 
+
+From Unity 2020.2, a new rectangular area shadow have been introduce EvaluateShadow_RectArea and the function GetAreaLightAttenuation() have been rename to GetRectAreaShadowAttenuation(). Also the type DirectionalShadowType have been renamed SHADOW_TYPE.
```
Comment (JordanL8) [inline com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md:77]: From Unity 2020.2, HDRP includes a new rectangular area shadow, EvaluateShadow_RectArea. The GetAreaLightAttenuation() function has been renamed to GetRectAreaShadowAttenuation(). Also the type DirectionalShadowType have been renamed SHADOW_TYPE.

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md:77]: thanks

File: com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md (hunk 2 above)
Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md:44]: thanks

**Pull request merged**